### PR TITLE
doc: Adds 1.19.0 release upgrade guide

### PR DESCRIPTION
**Pull request by maastha**
## Description

Adds 1.19.0 release upgrade guide & adds a section for Terraform modules in our provider home page.

Link to any related issue(s): [CLOUDP-271231](https://jira.mongodb.org/browse/CLOUDP-271231)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: docs/guides/1.19.0-upgrade-guide.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+page_title: "Upgrade Guide 1.19.0"
+---
+
+# MongoDB Atlas Provider 1.19.0: Upgrade and Information Guide
+
+The Terraform MongoDB Atlas Provider version 1.19.0 has a number of new and exciting features.
+
+**New Resources, Data Sources, and Features:**
+- You can now [manage customer keys from Azure Key Vault over Private Endpoints](https://www.mongodb.com/docs/atlas/security/azure-kms-over-private-endpoint/#manage-customer-keys-with-azure-key-vault-over-private-endpoints) to further encrypt your data at rest in Atlas with the new `mongodbatlas_encryption_at_rest_private_endpoint` resource and data sources in conjunction with the existing `mongodbatlas_encryption_at_rest` resource.
+    - In order to configure a private endpoint for your Azure Key Vault, the corresponding `mongodbatlas_encryption_at_rest` resource has to be adjust by configuring `azure_key_vault_config.require_private_networking` to `true`. This attribute can be updated in place, ensuring the customer-managed keys encryption is never disabled.
```
Comment (AgustinBettati) [inline docs/guides/1.19.0-upgrade-guide.md]:
```suggestion
    - In order to configure a private endpoint for your Azure Key Vault, the corresponding `mongodbatlas_encryption_at_rest` resource has to be adjusted by configuring `azure_key_vault_config.require_private_networking` to `true`. This attribute can be updated in place, ensuring the customer-managed keys encryption is never disabled.
```

File: docs/guides/1.19.0-upgrade-guide.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+---
+page_title: "Upgrade Guide 1.19.0"
```
Comment (AgustinBettati) [inline docs/guides/1.19.0-upgrade-guide.md:2]: As mentioned by Zuhair we can add a small note on the publishing of new terraform modules. Not a provider feature precisely, but we can mention as a side note.

Comment (maastha) [reply docs/guides/1.19.0-upgrade-guide.md:2]: good point, let me add

Comment (maastha) [reply docs/guides/1.19.0-upgrade-guide.md:2]: added

File: docs/guides/1.19.0-upgrade-guide.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    - In order to configure a private endpoint for your Azure Key Vault, the corresponding `mongodbatlas_encryption_at_rest` resource has to be adjusted by configuring `azure_key_vault_config.require_private_networking` to `true`. This attribute can be updated in place, ensuring the customer-managed keys encryption is never disabled.
+    - To learn more, please review `mongodbatlas_encryption_at_rest_private_endpoint` [resource documentation](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/encryption_at_rest_private_endpoint).
+
+- You can now use the new `mongodbatlas_project_ip_addresses` data source that returns the IP addresses in an Atlas project categorized by services.
+
+- You can now manage [Atlas Stream Processors](https://www.mongodb.com/docs/atlas/atlas-stream-processing/overview/) with the new `mongodbatlas_stream_processor` resource, `mongodbatlas_stream_processor` and `mongodbatlas_stream_processors` data sources. To learn more, please review `mongodbatlas_stream_processor` [resource documentation](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/stream_processor).
```
Comment (oarbusi) [inline docs/guides/1.19.0-upgrade-guide.md:16]: second link does not yet exists, but will once the documentation for 1.19.0 is published in the registry

**Review by oarbusi — APPROVED** [review]

**Review by davidhou17 — APPROVED** [review]

**Review by lantoli — APPROVED** [review]

**Review by AgustinBettati — APPROVED** [review]
LGTM

File: docs/index.md [hunk 4]
```diff
@@ -225,3 +225,6 @@ in our GitHub repo that will help both beginner and more advanced users.
 Have a good example you've created and want to share?
 Let us know the details via an [issue](https://github.com/mongodb/terraform-provider-mongodbatlas/issues)
 or submit a PR of your work to add it to the `examples` directory in our [GitHub repo](https://github.com/mongodb/terraform-provider-mongodbatlas/).
+
+## Terraform MongoDB Atlas Modules
+You can now leverage our [Terraform Modules](https://registry.terraform.io/namespaces/terraform-mongodbatlas-modules) to easily get started with MongoDB Atlas and critical features like [Push-based log export](https://registry.terraform.io/modules/terraform-mongodbatlas-modules/push-based-log-export/mongodbatlas/latest), [Private Endpoints](https://registry.terraform.io/modules/terraform-mongodbatlas-modules/private-endpoint/mongodbatlas/latest), etc.
```
Comment (EspenAlbert) [inline docs/index.md:230]: I would prefer to explicitly mention all terraform modules and use a bullet list instead (the duplication in the upgrade-guide can stay the same IMO)

**Review by EspenAlbert — APPROVED** [review]

File: docs/index.md (hunk 4 above)
Comment (maastha) [reply docs/index.md:230]: I think that would bring additional maintenance for us to keep this doc & the modules in sync.
That's why I've purposely kept it generic

Comment (EspenAlbert) [reply docs/index.md:230]: IMO: Worth the cost, since we are not adding modules that often. (Feel free to resolve)

Comment (maastha) [reply docs/index.md:230]: It's not about the cost of adding new items to the list, it's more that this is something very easy to forget about, and can easily drift away from the actual list of modules, so I'd prefer to not include that dependency.

**Pull request merged**